### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by alinalaktionova**
Hi, @didevgen .
Please, check our last version of project

### CODE REVIEW

File: src/SmartHouse/SmartHouse.js [hunk 1]
```diff
@@ -1,103 +1,136 @@
...
-    const regex = /[\w\d\s]{5,10}/;
-    const result = name.match(regex);
-    if (result != null) {
-      Logger.warning("Name must contain 5-10 characters");
-      return false;
-    }
-    let isNameUnic = this._devices.find(device => {
+    return false;
+  });
+};
```
Comment (didevgen) [inline src/SmartHouse/SmartHouse.js]: return device instanceof obj

File: src/SmartHouse/SmartHouse.js [hunk 2]
```diff
@@ -1,103 +1,136 @@
...
+
+SmartHouse.prototype.getDeviceByName = function(name) {
+  if (typeof name === "string" && name.length) {
+    let device = this._devices.find(device => {
       if (device.getName() === name) {
         return true;
       }
```
Comment (didevgen) [inline src/SmartHouse/SmartHouse.js]: same as above

File: src/SmartHouse/SmartHouse.js [hunk 3]
```diff
@@ -1,103 +1,136 @@
...
-  this.addDevice = function(device) {
-    if (device instanceof Device) {
-      this._devices.push(device);
-    } else {
-      Logger.error("Devices must be objects of iKettle or Speaker");
+SmartHouse.prototype._isNameUnic = function(name) {
+  const result = this._devices.find(device => {
+    if (device.getName() === name) {
+      return true;
     }
```
Comment (didevgen) [inline src/SmartHouse/SmartHouse.js]: saem as above

File: src/SmartHouse/SmartHouse.js [hunk 4]
```diff
@@ -1,103 +1,136 @@
...
+  });
+  this._devices = result;
+};
 
-  this.addDevice = function(device) {
-    if (device instanceof Device) {
-      this._devices.push(device);
-    } else {
-      Logger.error("Devices must be objects of iKettle or Speaker");
+SmartHouse.prototype._isNameUnic = function(name) {
```
Comment (didevgen) [inline src/SmartHouse/SmartHouse.js]: unique

File: src/SmartHouse/SmartHouse.js [hunk 5]
```diff
@@ -1,103 +1,136 @@
...
-        return index;
-      }
-    });
+SmartHouse.prototype._checkName = function(name) {
+  if (typeof name !== "string") {
+    Logger.error("Name must be a string");
+    return false;
+  }
+  name = name.trim();
+  const regex = /[\w\d\s]{5,10}/;
```
Comment (didevgen) [inline src/SmartHouse/SmartHouse.js]: move to the shared file with constants

File: src/iKettle/iKettle.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { Device } from "../BaseDevices/Device/Device";
 import { Logger } from "../Utilities/Logger/Logger";
 
-export const iKettle = function(name, model = "iKettle") {
-  Device.call(this, name, model);
+export const iKettle = function(name) {
```
Comment (didevgen) [inline src/iKettle/iKettle.js]: iKettle? :)))

Comment (didevgen) [reply src/iKettle/iKettle.js]: Jobs is spinning in the coffin

**Review by didevgen — APPROVED** [review]

**Comment by danill1278** [comment]
Hi @didevgen, please take a look at changes.

**Pull request closed without merging**